clash-cores/src/Clash/Cores/Xilinx/Floating/Annotations.hs

```diff
@@ -126,3 +132,96 @@ veriBinaryPrim primName tclTFName funcName =
           format: Haskell
           templateFunction: #{tclTFName}
     |]
+
+vhdlFromUPrim
+  :: Name
+  -> String
+  -> Primitive
+vhdlFromUPrim primName funcName =
+  let tfName = 'fromUTclTF
+  in InlineYamlPrimitive [VHDL] [__i|
+    BlackBox:
+      name: #{primName}
+      type: |-
+        #{primName}
+          :: ( KnownDomain dom            -- ARG[0]
+             , KnownNat d                 -- ARG[1]
+             , HasCallStack               -- ARG[2]
+             )
+          => Clock dom                    -- ARG[3]
+          -> Enable dom                   -- ARG[4]
+          -> DSignal dom n (Unsigned ..)  -- ARG[5]
```

Comment:
Instead of doing this, you could move this into the where clause:

```
let
  knownDomArg = 0
  knownNatArg = 1
  ...
in
  ...
```

And then splice these number in, e.g.

```
~ARG[#{inputArg}]
```

This would make the code much easier to change if we ever need to add/remove an argument, not to mention easier to read. Perhaps you can do something similar for `~SYM[3]`.

What do you think?

Reply:
Also, isn't the result missing from the type signature?

Reply:
Oh, I like that idea!

And yes, the result is missing, this is just a cheat sheet for the correct numbers, and the result is `~RESULT`, no need for a cheat sheet. The code isn't correct Haskell anyway, I elided an unknown number with `..` (yeah, it's known here, it's 32, but I designed it to also work for `Unsigned 64` which I'll add later).

I would like to omit any unused variables from the assignments, though, so just `clockArg`, `enableArg` and `inputArg`. Otherwise you'll have useless code churn later when it were to change.

Reply:
How do you like it now?